### ISSUE TITLE
US-SDM: SDM-completion-wrapper-issue (was: 'Completion of target limits')

### ISSUE THREAD

**Issue by BerndHekele** [openETCS/modeling#313]
Current Implementation does not include special cases like
 - level crossing
 - release speed following fully the standard

**Comment by T12z** [comment]
The level crossing operator has been implemented, but is missing overall integration as there are no real interfaces for LX-information.
Release-speed-onboard calculation (3.13.9.4.8) has been implemented up to ~80%; 3.13.9.4.9 (limitation by MRSPs while in release speed monitoring) still missing.
In 3.13.9.4.8.2 (all formulars) are some awkward formulations. The Location of the EOA seems not to be absolute, in contrast to other locations, but related to the last balise group. The other problem is, that the calculation is iterative, but the cycle count is not given, nor limited. So I will have to take a wild guess: 10.
I will probably commit this state and then step to R16 and continue work.

**Comment by T12z** [comment]
blocked by #154

**Comment by JakobGartner** [comment]
again, we should talk (or meet)

**Comment by BaseliyosJacob** [comment]
No progress on issue #154 @jakobgartner

**Comment by BaseliyosJacob** [comment]
No progress on issue #154 . @jakobgartner will solve this week.

**Comment by BaseliyosJacob** [comment]
- [x] Analysis
- [x] Modelling
- [ ] Documentation
- [ ] Handover to V&V
- [ ] Approval from V&V

**Comment by T12z** [comment]
@BaseliyosJacob The hand-over work-flow should be shifted to a separate issue. The initial cause has been solved. The current impediment is rather "completion of speed and distance monitoring" which is still missing
 * brake models for lambda trains,
   - :point_right: though this is about to be provided in the next week/days
 * brake models for gamma trains,
   - :point_right: we will have to see how urgently this needs to be implemented, as there are other issues too and I am running out of capacity
 * A_gradient,
   - :point_right: Benjamin proposed to do it, was pulled off to another project too quickly (mind you, we have only one person officially on the project)
   - :point_right: as this is quite embedded in our package and its tangled structures, it may not be easily out-sourced to a helping hand and we will have to see when to finish this
 * proper integration of interfaces to TCO and Brake-feedback,
   - :point_right: what was actually the decision on integration of service-brake-feedback?

There is one internal issue on the target-limits-sub-package that is itiching in the back of my mind, that is about proper sorting of the EOA as a target compared to MRSP and LOA. But that again would be a bug and not about completion.

:fast_forward: long story short, I propose to close this issue and split it up into a monitoring completion issue and one or the other bug as needed.

cheers, Thorsten

**Comment by T12z** [comment]
@BaseliyosJacob please put Hand-Over work-flow marks where they fit or create an umbrella issue. Also I am not sure which issues needs completion for hand-over and for which release. Once this is handled, then please close the issue.

**Comment by T12z** [comment]
as the actual topic is long complete and the issue has been dragged to other things, I'll use this as a log into my issues. Next up:
 * [ ] work on documentation #716 
 * [ ] work on #658
 * support WP4
 * publishing results

**Comment by T12z** [comment]
Note to self: (known bugs that should be wasted some time onto)
 * calcDriverOutput -> check assignment of v_mrdt / v_target
 * SDM does not throw EB on obviously invalid input
 * SDM cannot run opposite direction -> throw brakes! instead of misbehaving
 * finish TCO / brake-FB

**Comment by T12z** [comment]
comment on last commit: exception handler has been introduced on input to exempt on invalid input and not implemented features that may lead to misbehaviour.

**Comment by T12z** [comment]
outstanding issues of interest, that are not followed by asignees:
[ ] #522 #794 usage of floats
[ ] #685 Gradient clean up
[ ] #657 Rotating mass
[ ] #541 Direction Controller --> MovementSupervision

**Comment by T12z** [comment]
next update,
so I closed implementation of adhesion functions as *wont fix* - it is just not worth for current releases.
* [ ] #658 could still receive some attention, if it is worth touching next weeks
* Missing TIU-data: M_rotating_nom (optional), whether TCO-command available, T_traction_cut_off, whether SB-command available (for #658, #657 )

**Comment by BerndHekele** [comment]
I close this issue.
When we go for the next round we will collect new features in SDM basedon the requirements.
